_posts/2023-04-21-extensibility-roadmap.markdown

```diff
@@ -0,0 +1,154 @@
+
+Practically speaking, in OpenSearch, we will need to solve for the ability for a subset of extensions to run in the same JVM, and operate on data structures without serialization or deserialization, all without the need to change anything in the implementation of the extension itself. 
+
+### Support dependencies between extensions
+
+In the current state all dependencies are implicit and all versions must match, and we heavily rely on testing a distribution bundle. There’s no mechanism to know what the dependencies are, and any dependency errors are found at runtime. Any update to the dependencies requires rebuilding another plugin, even when there are no changes within the current plugin. Thus, everything is always rebuilt from scratch for every release. To solve this, we will extend the ability for extensions to depend on other extensions, similar to the ability of an extension to depend on a semver-compatible version of OpenSearch/Dashboards.
+
+### Public catalog
+
+We’ll augment the previously built rudimentary schema for extension’s metadata to provide additional fields beyond name, version or compatibility. These will include such fields as well-defined categories, or additional vendor/sponsor information. We will build a minimal catalog website with search functionality, deploy and maintain a public version of it. We’ll ensure that the catalog system can also be run by any enterprise internally and build signing and trust into the system, but will not do any validation beyond metadata correctness at this stage. An internal catalog will be able to sync with a public catalog with appropriate administrative controls, or existing third party catalog systems can use an API to import extensions. Developers will be able to sign up to publish extensions on the public instance and we will build mechanisms that help users trust publishers. An API in OpenSearch will supporting installing extensions from catalogs. A view in OpenSearch Dashboards will allow browsing available catalogs and extensions.
+
+Developers will be able to publish an extension into a public catalog, users can find extensions in a catalog via a taxonomy and search functions, detail pages with meaningful metadata and vendor information, administrators can install extensions from public or private catalogs, and import a subset of a public catalog in their Enterprise. Finally, everyone we'll establish trust into who each extension publisher is claiming to be, and some quality controls and publisher approvals.
```

Comment:
This sentence doesn't make sense to me....

"Finally, everyone we'll establish trust into who each extension publisher is claiming to be, and some quality controls and publisher approvals."